crates/wasm-smith/Cargo.toml

```diff
@@ -21,6 +21,7 @@ flagset = { version = "0.4", features = ["serde"] }
 leb128 = "0.2.4"
 wasm-encoder = { version = "0.8.0", path = "../wasm-encoder" }
 indexmap = "1.6"
+serde = { version = "1", features = ['derive'] }
```

Comment:
Can we

* make this an optional dependency,
* make the `flagset` dependency's `"serde"` feature disabled by default,
* add a `"_internal_cli"` feature that enables both previous items,
* and then the CLI tool would enable this feature

This way, regular library usage doesn't need to pull in serde stuff here.